### PR TITLE
Add visual regression tests for dashboard page

### PR DESCRIPTION
**Pull request by gusaiani**
Adds visual tests to basic states of the Dashboard.

### CODE REVIEW

**Comment by gusaiani** [comment]
@metabase-bot run visual tests

**Comment by gusaiani** [comment]
@metabase-bot run visual tests

**Comment by gusaiani** [comment]
@metabase-bot run visual tests

**Comment by gusaiani** [comment]
@metabase-bot run visual tests

File: frontend/test/metabase-visual/dashboard/filter-sidebar.cy.spec.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { restore } from "__support__/e2e/cypress";
+
+const questionDetails = {
+  query: { "source-table": 2 },
```
Comment (nemanjaglumac) [inline frontend/test/metabase-visual/dashboard/filter-sidebar.cy.spec.js:4]: Let's import `SAMPLE_DATASET` and get `ORDERS_ID` from there instead of hard coding it, please.
```suggestion
  query: { "source-table": ORDERS_ID },
```

File: frontend/test/metabase-visual/dashboard/filter-sidebar.cy.spec.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  it("renders correctly", () => {
+    cy.createQuestionAndDashboard({ questionDetails }).then(
+      ({ body: { card_id, dashboard_id } }) => {
+        cy.visit(`/dashboard/${dashboard_id}`);
+
+        openFilterSidebar();
+
+        cy.findByText("Label");
+
+        cy.percySnapshot("Shows dashboard filter sidebar " + Date.now());
```
Comment (nemanjaglumac) [inline frontend/test/metabase-visual/dashboard/filter-sidebar.cy.spec.js:22]: We can easily leave only this part in the test itself and move the setup in the beforeEach hook.

File: frontend/test/metabase-visual/dashboard/fullscreen.cy.spec.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
+import { restore } from "__support__/e2e/cypress";
+import { SAMPLE_DATASET } from "__support__/e2e/cypress_sample_dataset";
+
+const { PRODUCTS, PRODUCTS_ID } = SAMPLE_DATASET;
+
+const questionDetails = {
+  name: "15119",
```
Comment (nemanjaglumac) [inline frontend/test/metabase-visual/dashboard/fullscreen.cy.spec.js:7]: Please remove the reference to the issue 15119 to avoid the confusion in the future. Thanks

**Review by nemanjaglumac — APPROVED** [review]
Left some inlined comments that need to be addressed, but otherwise looks good :)

File: frontend/test/metabase-visual/dashboard/filter-sidebar.cy.spec.js (hunk 1 above)
Comment (gusaiani) [reply frontend/test/metabase-visual/dashboard/filter-sidebar.cy.spec.js]: 👍 Done

File: frontend/test/metabase-visual/dashboard/filter-sidebar.cy.spec.js (hunk 2 above)
Comment (gusaiani) [reply frontend/test/metabase-visual/dashboard/filter-sidebar.cy.spec.js]: 👍 Done

File: frontend/test/metabase-visual/dashboard/fullscreen.cy.spec.js (hunk 3 above)
Comment (gusaiani) [reply frontend/test/metabase-visual/dashboard/fullscreen.cy.spec.js]: 👍 Done

File: frontend/test/metabase-visual/dashboard/fullscreen.cy.spec.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      },
+    );
+  });
+
+  it("renders in day mode and night mode", () => {
+    cy.icon("expand").click();
+
+    cy.icon("moon");
+
+    cy.percySnapshot("Shows dashboard in fullscreen day mode" + Date.now());
```
Comment (alxnddr) [inline frontend/test/metabase-visual/dashboard/fullscreen.cy.spec.js:58]: Why did you add Date.now? I'm asking because Percy compares screenshots with the same names so if it is dynamic they will be considered as always newly added screenshots

Comment (gusaiani) [reply frontend/test/metabase-visual/dashboard/fullscreen.cy.spec.js:58]: @alxnddr I had been playing with this in a personal Percy project, to learn how to leverage it before hitting CI, and for some reason it asked for a unique name.

Will remove.

Comment (alxnddr) [reply frontend/test/metabase-visual/dashboard/fullscreen.cy.spec.js]: If you have a single test that has multiple calls of `percySnapshot` — the last one will take the name combined name of contexts + test name. In your case `visual tests > dashboard > fullscreen renders in day mode and night mode`
For multiple calls in the same test you can use
``` 
cy.percySnapshot('night');
...
cy.percySnapshot('day');
```

**Comment by gusaiani** [comment]
@metabase-bot run visual tests

File: frontend/test/metabase-visual/dashboard/fullscreen.cy.spec.js (hunk 4 above)
Comment (gusaiani) [reply frontend/test/metabase-visual/dashboard/fullscreen.cy.spec.js]: @alxnddr 

Ah gotcha, makes sense. Updated.

Thank you for teaching me.

**Comment by gusaiani** [comment]
@metabase-bot run visual tests

**Review by alxnddr — APPROVED** [review]
